.github/workflows/fast-forward.yml

```diff
@@ -7,6 +7,9 @@ name: Fast-forward tracking branch for selected CodeQL version
 on:
   workflow_dispatch:
 
+permissions:
+  contents: write
+
```

Comment:
Can we remove the permissions block in the `fast-forward` job below as it's redundant?